crates/holochain/src/core/ribosome/real_ribosome.rs

```diff
@@ -737,4 +772,16 @@ pub mod wasm_test {
             unreachable!();
         }
     }
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn the_incredible_halt_test() {
+        observability::test_run().ok();
+        let RibosomeTestFixture {
+            conductor, alice, ..
+        } = RibosomeTestFixture::new(TestWasm::TheIncredibleHalt).await;
+
+        // This will run infinitely unless our metering kicks in and traps it.
+        let result: Result<(), _> = conductor.call_fallible(&alice, "smash", ()).await;
```

Comment:
Shouldn't this have a timeout?

Reply:
why would we have it timeout? if it runs forever that means metering is broken, which needs to be fixed, not timed out and ignored

Reply:
or, to put it another way, the metering _is_ the timeout, measured in ops rather than seconds

Reply:
TLDR: +1 on explicit timeout and error signalling here

i see @freesig's point here. we could timeout and fail the test explicitly with an error like "the wasm operation limit did not halt the call within X time".

if we don't do this we let the testrunner handle the timeout, which, even though ours is now configured to do that, i think is a semantic leak because that mechanism was designed to work backwards on detecting cases where the test cases weren't foreseen to run too long or freeze.